google/cloud/storage/tests/object_integration_test.cc

```diff
@@ -141,6 +141,44 @@ TEST_F(ObjectIntegrationTest, ListObjects) {
   }
 }
 
+TEST_F(ObjectIntegrationTest, ListObjectsVersions) {
+  auto bucket_name = ObjectTestEnvironment::bucket_name();
+  Client client;
+  // Make sure the Bucket is configured to have multiple versions of each
+  // object.
+  client.PatchBucket(bucket_name, BucketMetadataPatchBuilder().SetVersioning(
```

Comment:
Good catch, thanks!  Since we always use the same bucket for these tests, and it is already configured with versioning I changed the code to just check that this is the case.
Creating a new bucket is also possible (we do that in the bucket integration tests), but seemed kind of overkill here.

Reply:
Because the propagation delay is so small and happens very rarely, I haven't been able to confirm whether object deletions might still follow the old versioning setting, despite returning {versioning: true} in the result of a GetBucketMetadata call.  That being said, I think it's pretty unlikely that we'll hit this flake if we're performing a few operations between when versioning is set and when we actually need to rely on that behavior... But I think it's worth adding a note where we create the bucket and set versioning that it may take up to 30 seconds (but usually takes less than a second) for versioning behavior to take effect (and link to https://cloud.google.com/storage/docs/object-versioning).